server/fishtest/rundb.py

```diff
@@ -427,10 +427,9 @@ def scavenge(self, run):
         # print("scavenge ", run["_id"])
         dead_task = False
         old = datetime.utcnow() - timedelta(minutes=6)
-        task_id = -1
         run_id = str(run["_id"])
-        for task in run["tasks"]:
-            task_id += 1
+        for task_id in list(run["tasks"]):
```

Comment:
https://github.com/glinscott/fishtest/blob/16d52ff19438f7af513ddf17df633dc9ebad314f/server/fishtest/rundb.py#L184-L194

`run["tasks"]` is already a list. In the proposed code `task_id` is one of the tasks (a dict with all the info about the task), so the code should break.

Reply:
This is not what I was proposing....